src/zonetree/answer.rs

```diff
@@ -0,0 +1,217 @@
+//! Answers to zone tree queries.
+
+//------------ Answer --------------------------------------------------------
+
+use super::{SharedRr, SharedRrset, StoredDname};
+use crate::base::iana::Rcode;
+use crate::base::message_builder::AdditionalBuilder;
+use crate::base::wire::Composer;
+use crate::base::Message;
+use crate::base::MessageBuilder;
+use octseq::Octets;
+
+/// A DNS answer to a query against a [`Zone`].
+///
+/// [`Answer`] is the type returned by [`ReadableZone::query()`].
+///
+/// Callers of [`ReadableZone::query()`] will likely only ever need to use the
+/// [`Self::to_message()`] function. Alternatively, for complete control use
+/// the getter functions on [`Answer`] instead and construct a response
+/// message yourself using [`MessageBuilder`].
+///
+/// Implementers of alternate backing stores for [`Zone`]s will need to use
+/// one of the various `Answer` constructor functions when
+/// [`ReadableZone::query()`] is invoked for your zone content in order to
+/// tailor the DNS message produced by [`Self::to_message()`] based on the
+/// outcome of the query.
+///
+/// [`Zone`]: crate::zonetree::Zone
+/// [`ReadableZone::query()`]: crate::zonetree::traits::ReadableZone::query()
+#[derive(Clone)]
+pub struct Answer {
```

Comment:
I think because it's supposed to represent a limited "answer" to a zone (tree) query only, not an entire DNS message. `to_message()` turns it into the start of a possible DNS message to send back to the client, but doesn't for example set things like AA (as it doesn't know if the application is truly authoritative for the zone contents I suppose - this relates to a comment you make below about the missing AA flag). Also there isn't currently any logic in `to_message()` to add anything to an additional sectio. @partim may be able to say more about this.

Reply:
It could also be because the way the builder works the caller wouldn't know what had already been added to the additional section and would have trouble therefore adding its own stuff to it, could that be part of it?

Reply:
The intention was to have the additional data as part of the `SharedRrset` in the `content`. That just hasn’t happened yet.

Reply:
Added to the TODO list.